src/main/java/com/minecolonies/colony/workorders/WorkOrderBuild.java

```diff
@@ -51,9 +57,21 @@ public WorkOrderBuild(@NotNull AbstractBuilding building, int level)
         this.buildingLocation = building.getID();
         this.upgradeLevel = level;
         this.upgradeName = building.getSchematicName() + level;
-        this.schematicName = building.getStyle() + '/' + this.getUpgradeName();
         this.buildingRotation = building.getRotation();
         this.cleared = level > 1;
+
+        try
+        {
+            SchematicFormat.readFromStream(
+                    SchematicWrapper.getStream(
+                            new ResourceLocation("minecolonies:schematics/" + building.getStyle() + '/' + this.getUpgradeName() + ".schematic")));
+            this.schematicName = building.getStyle() + '/' + this.getUpgradeName();
+        }
+        catch (RuntimeException e)
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1166)